image_pool.py

```diff
@@ -0,0 +1,53 @@
+from directory_walker import DirectoryWalker
+from progress_bar import progress_bar
+import color_spaces as cs
+import Image
+from image_functions import *
+import logging
+
+# Configure logger.
+FORMAT = "%(name)s.%(funcName)s:  %(message)s"
+logging.basicConfig(level=logging.INFO, format=FORMAT)
+logger = logging.getLogger(__name__)
+
+class ImagePool(dict):
```

Comment:
I like the way you use `__contains__` and `__len__` in the subclass, but I'm not sure why this should subclass dict. I can't figure out what `__getitem__` and `__setitem__` should do to an ImagePool, so I think it probably isn't a dict; it's just an object that implements `__contains__` and `__len__`.